# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/IPCUtils.h
# Commit: bb63792c3333
# Full Hash: bb63792c3333a7911bdfad9e9e9ec58bb48bad93
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-08 04:19:40
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 2: Create PNotification via PBackground r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227687
# ==============================================================================

diff -r 93937caf6d13 -r bb63792c3333 dom/notification/IPCUtils.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/notification/IPCUtils.h	Thu Nov 07 17:32:03 2024 +0000
@@ -0,0 +1,29 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef DOM_NOTIFICATION_IPCUTILS_H_
+#define DOM_NOTIFICATION_IPCUTILS_H_
+
+#include "ipc/EnumSerializer.h"
+#include "ipc/IPCMessageUtils.h"
+#include "mozilla/dom/NotificationBinding.h"
+
+namespace IPC {
+
+using NotificationDirection = mozilla::dom::NotificationDirection;
+template <>
+struct ParamTraits<NotificationDirection>
+    : public ContiguousEnumSerializerInclusive<NotificationDirection,
+                                               NotificationDirection::Auto,
+                                               NotificationDirection::Rtl> {};
+
+DEFINE_IPC_SERIALIZER_WITH_FIELDS(mozilla::dom::NotificationBehavior, mNoclear,
+                                  mNoscreen, mShowOnlyOnce, mSoundFile,
+                                  mVibrationPattern);
+
+}  // namespace IPC
+
+#endif  // DOM_NOTIFICATION_IPCUTILS_H_